frontend/controller/service-worker.js

```diff
@@ -183,72 +183,83 @@ sbp('sbp/selectors/register', {
   // In theory, the `PushManager` APIs used are available in the SW and we could
   // have this function there. However, most examples perform this outside of the
   // SW, and private testing showed that it's more reliable doing it here.
-  'service-worker/setup-push-subscription': async function (retryCount?: number) {
-    if (process.env.CI) {
+  'service-worker/setup-push-subscription': async function (attemptNumber = 1) {
+    if (window.Cypress || process.env.CI) {
```

Comment:
Not a strong feeling either way, but the reason I was _not_ checking for `window.Cypress` (and generally avoid doing so) is that these are runtime checks (code shipped) while `process.env.CI` is exclusively a buildtime check (no code shipped).

Reply:
Had to add it because without it the tests weren't passing. An alternative would be to create a Cypress envvar in `Gruntfile.js`...